src/lib/useQueryParams.ts

```diff
@@ -32,15 +28,17 @@ export default function useQueryParams(
   const location = useLocation();
   const navigate = useNavigate();
   const origParams = queryString.parse(location.search, options);
-  const setQueryParam = (key: string, value: any, replace = false) => {
-    if (value === undefined) {
-      delete origParams[key];
-    } else {
-      origParams[key] = value;
-    }
+
+  const setQueryParams = (params: Record<string, any>, replace = false) => {
+    Object.keys(params).forEach((k) => {
+      if (params[k] === undefined) {
+        delete params[k];
+      }
+    });
+
     navigate(
       {
-        search: queryString.stringify(origParams, options),
+        search: queryString.stringify(params, options),
```

Comment:
This'll end up overwriting everything in the query string. You need to modify origParams instead unless you intend to overwrite the whole query string.